packages/sdk/tests/e2e/dataProtectorSharing/subscribeToCollection.test.ts

```diff
@@ -45,4 +45,45 @@ describe('dataProtector.subscribe()', () => {
         timeouts.subscribe
     );
   });
+
+  describe('When calling subscribe() with invalid collection ID', () => {
+    it(
+      'should throw the corresponding error',
+      async () => {
+        const collectionTokenId = -1;
+
+        // --- WHEN / THEN
+        await expect(
+          dataProtectorEndUser.sharing.subscribeToCollection({
+            collectionTokenId,
+          })
+        ).rejects.toThrow(
+          new Error('collectionTokenId must be greater than or equal to 0')
+        );
+      },
+      timeouts.subscribe
+    );
+  });
+
+  describe('When calling subscribe() with collection ID that does not exist', () => {
+    it(
+      'should throw the corresponding error',
+      async () => {
+        // Increment this value as needed
+        const collectionTokenIdThatDoesNotExist = 9999999;
+
+        // --- WHEN / THEN
+        await expect(
+          dataProtectorEndUser.sharing.subscribeToCollection({
+            collectionTokenId: collectionTokenIdThatDoesNotExist,
```

Comment:
After v2 merging. You should add duration now, in order to avoid the collection owner to front run your transaction and change the subscription duration.